contrib/gogearbox/gearbox.v1/gearbox.go

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package gearbox provides functions to trace the gogearbox/gearbox package (https://github.com/gogearbox/gearbox)
+package gearbox // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/gogearbox/gearbox.v1"
+
+import (
+	"fmt"
+	"strconv"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/internal/fasthttptrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+
+	"github.com/gogearbox/gearbox"
+	"github.com/valyala/fasthttp"
+)
+
+const componentName = "gogearbox/gearbox.v1"
+
+func init() {
+	telemetry.LoadIntegration(componentName)
+	tracer.MarkIntegrationImported(componentName)
+}
+
+// Middleware returns middleware that will trace incoming requests.
+func Middleware(opts ...Option) func(gctx gearbox.Context) {
+	cfg := newConfig()
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	log.Debug("contrib/gogearbox/gearbox.v1: Configuring Middleware: cfg: %#v", cfg)
+	spanOpts := []tracer.StartSpanOption{
+		tracer.ServiceName(cfg.serviceName),
+	}
+	return func(gctx gearbox.Context) {
+		if cfg.ignoreRequest(gctx) {
+			gctx.Next()
+			return
+		}
+		fctx := gctx.Context()
+		spanOpts = append(spanOpts, defaultSpanOptions(fctx)...)
+		// Create an instance of FasthttpCarrier, which embeds *fasthttp.RequestCtx and implements TextMapReader
```

Comment:
```suggestion
```

[nit] I think this comment can be omitted as its just describing what the code below is doing (as opposed as a comment that would explain why something is done)